tools/network/resolveController.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2019-2020 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package network
+
+import (
+	"net"
+	"time"
+
+	"github.com/algorand/go-algorand/logging"
+	"github.com/algorand/go-algorand/tools/network/dnssec"
+)
+
+// ResolverIf is re-import from dnssec.ResolverIf
+type ResolverIf interface {
+	dnssec.ResolverIf
+}
+
+// compile-time check
+var _ dnssec.ResolverIf = &net.Resolver{}
+var _ dnssec.ResolverIf = &Resolver{}
+var _ dnssec.ResolverIf = &dnssec.Resolver{}
```

Comment:
nit: add these compile-time checks to the `XXX_test.go` file so that they won't get instantiated on the "production" binary.

Reply:
they will be stripped out anyway since not used